### ISSUE TITLE
the rule explicit-length-check should be used in comparisons only

### ISSUE THREAD

**Issue by eng-dibo** [sindresorhus/eslint-plugin-unicorn#1840]
the rule `explicit-length-check` should be used in comparisons only. 
and not be used in other situations such as assigning a variable like this:

```
 let x =  options.length || 500
```
and this:
```
something( options.length || 500 )
```

also it should be linted only if attached to a string like this
```


function example( text: string, options: {[key:string]: any}){
// .length here is a prototype of string
  text.length

 // but here is a property of an object
  options.length
}
```

**Comment by sindresorhus** [comment]
> `let x = options.length || 500`

I disagree. The whole point of the rule is to be explicit. It should be `let x = options.length > 0 || 500`.

**Comment by sindresorhus** [comment]
> // but here is a property of an object
> options.length

The lint rule is not aware of types. Even then, most cases of a `.length` property is a number, so linting it is correct.

**Comment by sindresorhus** [comment]
> and not be used in other situations such as assigning a variable like this:

👍

**Comment by eng-dibo** [comment]
you may check that the context is inside an `if` statement or any similar statements.

**Comment by fregante** [comment]
I agree that this is a bug:

```js
const excuses = []
const pullups = excuses.length || 42;
```

I'm not sure if it ever makes sense though, OP’s example should trigger these instead:

- https://typescript-eslint.io/rules/prefer-nullish-coalescing/
- https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-default-parameters.md

**Comment by SamanthaAdrichem** [comment]
We have this issue in reverse and it's complaining about that it should be > 0

We have a shared component in which the one API call returns a paginated totalCount and the other just returns an array.
We use the totalCount if it exists and otherwise the length.
Is it perfect? no, Should it be improved yes.

But the current linting complains with explicit-length-check and says this line of code;

```
this.itemCount = result.totalCount || result.length;
```

Should become
```
this.itemCount = result.totalCount || result.length > 0;
```

Which is completely wrong.
It's not an if statement, it's not a check, it's an assignment.

We solved it by typecasting the length to its own type

**Comment by axetroy** [comment]
I got the same false positives and autofix has broken my program.

The source of the error is here

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d3e4b805da31c6ed7275e2e2e770b6b0fbcf11c2/rules/utils/boolean.js#L58-L62

I will try fix it later.
